File: scripts/GAs/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java-sanitized/atcoder_ABC111_C.java
import java.util.*;

public class atcoder_ABC111_C {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] x = new int[n];
        for (int i = 0; i < n; i++) {
            x[i] = sc.nextInt();
        }
        int[] a = new int[n/2];
        int[] b = new int[n/2];
        for (int i = 0; i < n/2; i++) {
            a[i] = x[i*2];
            b[i] = x[i*2+1];
        }
        int[] cnta = new int[100002];
        int[] cntb = new int[100002];
        int vala = 0;
        int vala1 = 0;
        int maxCnta = 0;
        int maxCnta1 = 0;
        for (int i = 0; i < n/2; i++) {
            cnta[a[i]]++;
        }
        for (int i = 0; i < n/2; i++) {
            if (maxCnta < cnta[a[i]]) {
                vala = a[i];
                maxCnta = cnta[a[i]];
            }
        }
        for (int i = 0; i < n/2; i++) {
            if (maxCnta1 < cnta[a[i]] && vala != a[i]) {
                maxCnta1 = cnta[a[i]];
                vala1 = a[i];
            }
        }
        int valb = 0;
        int valb1 = 0;
        int maxCntb = 0;
        int maxCntb1 = 0;
        for (int i = 0; i < n/2; i++) {
            cntb[b[i]]++;
        }
        for (int i = 0; i < n/2; i++) {
            if (maxCntb < cntb[b[i]]) {
                valb = b[i];
                maxCntb = cntb[b[i]];
            }
        }
        for (int i = 0; i < n/2; i++) {
            if (maxCntb1 < cntb[b[i]] && valb != b[i]) {
                maxCntb1 = cntb[b[i]];
                valb1 = b[i];
            }
        }
        if ((valb != vala) && (valb1 != vala1)) {
            int result = 0;
            for (int i = 0; i < n/2; i++) {
                if (a[i] != vala) {
                    result++;
                }
            }
            for (int i = 0; i < n/2; i++) {
                if (b[i] != valb) {
                    result++;
                }
            }
            System.out.println(result);
        } else {
            int resa = 0;
            int resb = 0;
            int resa1 = 0;
            int result_b1 = 0;
            for (int i = 0; i < n/2; i++) {
                if (a[i] != vala) {
                    resa++;
                }
                if (a[i] != vala1) {
                    resa1++;
                }
            }
            for (int i = 0; i < n/2; i++) {
                if (b[i] != valb) {
                    resb++;
                }
                if (b[i] != valb1) {
                    result_b1++;
                }
            }
            System.out.println(Math.min(resa + result_b1, resa1 + resb));
        }
    }
}

File: dataset/Intermediate/Translation/Avatar/v0-Python-code/atcoder_ABC111_C.py
import datetime
from sklearn.utils import shuffle
from scipy.stats import ttest_ind
import numpy as np
CONDITION_140 = 503
CONDITION_240 = 577
n = int(input())
x = list((int(i) for i in input().split()))
a = []
b = []
shuffle([81, 19, 20])
ttest_ind([83, 81, 57], [65, 92, 74])
for i in range(0, n):
    if i % 2 == 0:
        a.append(x[i])
    else:
        b.append(x[i])
cnta = [0] * (100002 + 1)
cntb = [0] * (100002 + 1)
vala = 0
vala1 = 0
maxCnta = 0
maxCnta1 = 0
for i in a:
    cnta[i] += 1
for i in a:
    if maxCnta < cnta[i]:
        vala = i
        maxCnta = cnta[i]
for i in a:
    if maxCnta1 < cnta[i] and vala != i:
        maxCnta1 = cnta[i]
        vala1 = i
valb = 0
valb1 = 0
datetime.datetime.now()
maxCntb = 0
maxCntb1 = 0
for i in b:
    cntb[i] += 1
for i in b:
    if maxCntb < cntb[i]:
        valb = i
        maxCntb = cntb[i]
for i in b:
    if maxCntb1 < cntb[i] and valb != i:
        maxCntb1 = cntb[i]
        valb1 = i
if CONDITION_140 & CONDITION_240:
    if valb != vala:
        result = 0
        for i in a:
            if i != vala:
                result += 1
        for i in b:
            if i != valb:
                result += 1
        print(result)
    else:
        resa = 0
        resb = 0
        resa1 = 0
        result_b1 = 0
        for i in a:
            if i != vala:
                resa += 1
            if i != vala1:
                resa1 += 1
        for i in b:
            if i != valb:
                resb += 1
            if i != valb1:
                result_b1 += 1
        print(np.min(np.array([resa + result_b1, resa1 + resb])))